src/main/java/io/aiven/kafka/connect/opensearch/spi/ClientsConfiguratorProvider.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 Aiven Oy
+ * Copyright 2016 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.opensearch.spi;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.ServiceLoader;
+
+import io.aiven.kafka.connect.opensearch.OpensearchSinkConnectorConfig;
+
+public final class ClientsConfiguratorProvider {
+    private ClientsConfiguratorProvider() {
+    }
+
+    /**
+     * Use {@link ServiceLoader} mechanism to discover available configurators for Opensearch and Kafka clients 
+     * which are applicable to the provided configuration. 
+     * @param config provided configuration
+     * @return the list of discovered {@link OpensearchClientConfigurator} configurators which are applicable to the 
+     *     provided configuration.
+     */
+    public static Collection<OpensearchClientConfigurator> forOpensearch(final OpensearchSinkConnectorConfig config) {
+        final Collection<OpensearchClientConfigurator> configurators = new ArrayList<>();
+        final ServiceLoader<OpensearchClientConfigurator> loaders = ServiceLoader
+            .load(OpensearchClientConfigurator.class);
```

Comment:
Due to [obscure issues with Connect class loading](https://issues.apache.org/jira/browse/KAFKA-12610), we should explicitly specify which class loader to use here:
```suggestion
            .load(OpensearchClientConfigurator.class, ClientsConfiguratorProvider.class.getClassLoader());
```

FWIW: I thought we'd fixed that issue a while go and I'll try to get a fix in sometime soon, but even so, it's best to preserve backwards compatibility between newer releases of this connector and older versions of Connect that won't have the fix.

Reply:
:+1: